### PR TITLE
Environment variables, readme and code improvements

### PR DESCRIPTION
**Pull request by Varun-garg**
Signed-off-by: Varun Garg <varun.ga@browserstack.com>

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "python.linting.pylintEnabled": true,
+  "python.pythonPath": "/usr/local/bin/python"
+}
```
Comment (francisf) [inline .vscode/settings.json]:
```suggestion
}

```

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "python.linting.pylintEnabled": true,
+  "python.pythonPath": "/usr/local/bin/python"
```
Comment (francisf) [inline .vscode/settings.json]: very os specific

File: README.md [hunk 3]
```diff
@@ -1,8 +1,10 @@
-automate-python-samples
-=======================
+# automate-python-samples
 
 Documentation for writing Automate test scripts in Python.
 
+## Environment variables
+To test varios sample repositories with ease,  it is recommended to setup `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` environment variables. If you don't wish you can directly edit the samples and add add credentials in them.
```
Comment (francisf) [inline README.md]:
```suggestion
To test varios sample repositories with ease,  it is recommended to setup `BROWSERSTACK_USERNAME` and `BROWSERSTACK_ACCESS_KEY` environment variables. Alternatively you can directly update the samples with the credentials or pass the appropriate command line parameters.
```

File: .vscode/settings.json (hunk 2 above)
Comment (Varun-garg) [reply .vscode/settings.json]: accidently added this file

File: README.md (hunk 3 above)
Comment (Varun-garg) [reply README.md]: could you also correct `varios`?

File: google-search-browserstack.py [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
-    BROWSERSTACK_ACCESS_KEY = sys.argv[2]
+    USERNAME = os.environ.get('BROWSERSTACK_USERNAME') or sys.argv[1]
+    BROWSERSTACK_ACCESS_KEY = os.environ.get('BROWSERSTACK_ACCESS_KEY') or sys.argv[2]
 except IndexError:
     print("Pleaes provide the username and browserstack access key as command line arguments.")
     sys.exit(1)
 
 driver = webdriver.Remote(
-    command_executor='http://%s:%s@hub.browserstack.com/wd/hub' %(
+    command_executor='http://%s:%s@hub.browserstack.com/wd/hub' % (
```
Comment (francisf) [inline google-search-browserstack.py]:
```suggestion
    command_executor='https://%s:%s@hub.browserstack.com/wd/hub' % (
```

File: nose-test/google-search-browserstack.py [hunk 5]
```diff
@@ -6,12 +7,11 @@
...
-USERNAME = None
-BROWSERSTACK_ACCESS_KEY = None
+USERNAME = os.environ.get('BROWSERSTACK_USERNAME') or sys.argv[3]
+BROWSERSTACK_ACCESS_KEY = os.environ.get('BROWSERSTACK_ACCESS_KEY') or sys.argv[4]
 
 if not (USERNAME and BROWSERSTACK_ACCESS_KEY):
     raise Exception("Please provide your BrowserStack username and access key")
-    sys.exit(1)
 
 class PythonOrgSearch(unittest.TestCase):
```
Comment (francisf) [inline nose-test/google-search-browserstack.py]: https on l19

File: parallel_tests/README.md [hunk 6]
```diff
@@ -17,37 +19,9 @@ browsers.json - This file specifies the browser capabilites for the tests. Here,
...
-    inputElement = driver.find_element_by_name("q")
-    inputElement.send_keys("browserstack")
-    inputElement.submit()
-    print driver.title
-    driver.quit()
-
-run_parallel_tests.py - This is the runner which runs the tests in parallel.  
+`test.py <browsers.json> <capability id>` - This file contains the selenium test which would be run in each of the browsers specificed by "browsers.json". 
+
+`run_parallel_tests.py <test.py> <browsers.json>` - This is the runner which runs the tests in parallel.
```
Comment (francisf) [inline parallel_tests/README.md]: nice!!

File: parallel_tests/test.py [hunk 7]
```diff
@@ -23,8 +25,10 @@
 # THE TEST TO BE RUN PARALLELY GOES HERE
 
 driver = webdriver.Remote(
-  command_executor='http://<browserstack_UserName>:<browserstack_AccessKeys>@hub.browserstack.com/wd/hub',
-  desired_capabilities=caps)
+    command_executor='http://%s:%s@hub.browserstack.com/wd/hub' % (
```
Comment (francisf) [inline parallel_tests/test.py]:
```suggestion
    command_executor='https://%s:%s@hub.browserstack.com/wd/hub' % (
```

File: py.test/google-search-browserstack.py [hunk 8]
```diff
@@ -1,13 +1,15 @@
...
-BROWSERSTACK_ACCESS_KEY = None
+USERNAME = os.environ.get('BROWSERSTACK_USERNAME') or sys.argv[1]
+BROWSERSTACK_ACCESS_KEY = os.environ.get('BROWSERSTACK_ACCESS_KEY') or sys.argv[2]
 
 if not (USERNAME and BROWSERSTACK_ACCESS_KEY):
     raise Exception("Please provide your BrowserStack username and access key")
-    sys.exit(1)
 
 def test_run():
     url = "http://%s:%s@hub.browserstack.com/wd/hub" %(
```
Comment (francisf) [inline py.test/google-search-browserstack.py]:
```suggestion
    url = "https://%s:%s@hub.browserstack.com/wd/hub" %(
```

File: unittest/google-search-browserstack.py [hunk 9]
```diff
@@ -6,8 +7,8 @@
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
 
 try:
-    USERNAME = sys.argv[1]
-    BROWSERSTACK_ACCESS_KEY = sys.argv[2]
+    USERNAME = os.environ.get('BROWSERSTACK_USERNAME') or sys.argv[1]
+    BROWSERSTACK_ACCESS_KEY = os.environ.get('BROWSERSTACK_ACCESS_KEY') or sys.argv[2]
 except IndexError:
     print("Please provide the username and browserstack access key as command line arguments.")
     sys.exit(1)
```
Comment (francisf) [inline unittest/google-search-browserstack.py]: l19 https

**Pull request merged**